security/pkg/nodeagent/secretfetcher/secretfetcher.go

```diff
@@ -330,11 +358,19 @@ func (sf *SecretFetcher) scrtUpdated(oldObj, newObj interface{}) {
 	}
 }
 
-// FindIngressGatewaySecret returns the secret for a k8sKeyA, or empty secret if no
+// FindIngressGatewaySecret returns the secret whose name matches the key, or empty secret if no
 // secret is present. The ok result indicates whether secret was found.
+// If ServeFallbackSecret is true, FallbackSecretName is specified and there is a secret named
+// FallbackSecretName, return the fall back secret.
 func (sf *SecretFetcher) FindIngressGatewaySecret(key string) (secret model.SecretItem, ok bool) {
 	val, exist := sf.secrets.Load(key)
 	if !exist {
+		if sf.ServeFallbackSecret {
+			fallbackVal, fallbackExist := sf.secrets.Load(sf.FallbackSecretName)
```

Comment:
To make fallback secret available in cache, does it mean customer need to start nodeagent first then create that fallback k8s secret in same ns ?

Reply:
Yes, customer need to create a fall back secret in the same namespace as the ingress gateway pod. That secret needs to be created before deploying Gateway, so that node agent could pick it early.

Reply:
what if after customer set fallback secret then nodeagent restart, will it still able to get the fallback secret ?